app/models/user.rb

```diff
@@ -80,6 +89,38 @@ def self.for_email(email)
     User.find_or_create_by(email_address: email.strip.downcase)
   end
 
+  def self.exists_with_role_slug?(role, slug)
+    the_role = role.is_a?(Role) ? role : Role.find_by_name(role)
+    return false unless the_role
+    triple_exists = false
+    the_role.users.each do |u|
+      if u.client_slug == slug
+        triple_exists = true
+        break
+      end
+    end
+    triple_exists
+  end
```

Comment:
Would it make sense to do this as a validation on the `Role`? (Legitimately asking.)